### PR TITLE
[SMALLFIX] Support getInodeList() for child descendant

### PR DESCRIPTION
**Pull request by ifcharming**

### CODE REVIEW

**Comment by ifcharming** [comment]
@gpang I've added unit tests for TempInodePathForDescendant

File: core/server/master/src/main/java/alluxio/master/file/meta/TempInodePathForDescendant.java [hunk 1]
```diff
@@ -87,7 +87,18 @@ public synchronized  InodeDirectory getParentInodeDirectory()
     if (mDescendantInode == null) {
       return super.getInodeList();
     }
-    throw new UnsupportedOperationException();
+    try {
+      if (super.getInode().getId() == mDescendantInode.getParentId()) {
+        // Only return the inode list if the descendant is the direct child.
```
Comment (gpang) [inline core/server/master/src/main/java/alluxio/master/file/meta/TempInodePathForDescendant.java]: If `getInodeList` can detect when it is valid, then should `getParentInodeDirectory` also detect it?

Comment (ifcharming) [inline core/server/master/src/main/java/alluxio/master/file/meta/TempInodePathForDescendant.java]: Good catch, I've added support for `getParentInodeDirectory`. thanks

**Comment by ifcharming** [comment]
@gpang I've addressed your comment. PTAL

File: core/server/master/src/main/java/alluxio/master/file/meta/TempInodePathForDescendant.java [hunk 2]
```diff
@@ -79,15 +80,39 @@ public synchronized  InodeDirectory getParentInodeDirectory()
     if (mDescendantInode == null) {
       return super.getParentInodeDirectory();
     }
-    throw new UnsupportedOperationException();
+    try {
+      if (super.getInode().getId() == mDescendantInode.getParentId()) {
+        Inode<?> parentInode = super.getInode();
```
Comment (gpang) [inline core/server/master/src/main/java/alluxio/master/file/meta/TempInodePathForDescendant.java]: NIT: Add a comment that the descendant is direct child, so getting parent is possible.

Comment (ifcharming) [inline core/server/master/src/main/java/alluxio/master/file/meta/TempInodePathForDescendant.java]: done

**Comment by ifcharming** [comment]
@gpang I've updated the PR, thanks.

**Pull request merged**